lib/publiccloud/ec2.pm

```diff
@@ -241,6 +241,36 @@ sub stop_instance
     die("Failed to stop instance $instance_id") unless ($attempts > 0);
 }
 
+sub suspend_instance {
+    my ($self, $instance) = @_;
+    my $instance_id = $instance->instance_id();
+    my $attempts = 60;
+
+    die("Outdated instance object") if ($instance->public_ip ne $self->get_public_ip());
```

Comment:
Why do we need to `die` here and not just update the public_ip? :thinking: 

```suggestion
    $instance->public_ip = $self->get_public_ip();
```

Reply:
True. Did so.

Reply:
I would prefer if logic which changing IP address would be in SINGLE place ( wait_for_ssh function and not split over all code ) generally speaking I don't know why do we care about IP here at all we suppose to check if instance stopped or not and because we doing it via API call anyway we not care about IP here . so let's simply drop this line